realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -367,7 +372,13 @@ private void completedAsyncRealmResults(QueryUpdateTask.Result result) {
                     if (!realmResults.isLoaded()) {
                         RealmLog.d("[COMPLETED_ASYNC_REALM_RESULTS "+ weakRealmResults + "] , realm:"+ HandlerController.this + " same versions, using results (RealmResults is not loaded)");
                         // swap pointer
-                        realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
+                        try {
+                            realmResults.swapTableViewPointer(result.updatedTableViews.get(weakRealmResults));
+                        } catch (BadVersionException e) {
```

Comment:
I switched BadVersionException to be Checked instead of runtime to make sure that all code paths where aware it might be a problem. In this case the handover from the caller -> worker did succeed, but I am still not confident that you can do worker -> caller safely as the worker has been closed at this point and the caller might have advanced.